crates/claims/crates/vc/src/syntax/non_empty_vec.rs

```diff
@@ -0,0 +1,90 @@
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(try_from = "Vec<T>", into = "Vec<T>")]
+pub struct NonEmptyVec<T: Clone>(Vec<T>);
```

Comment:
We don't want the `Clone` bound here. As you can see it propagates everywhere.

Reply:
Yeah, but I'm not sure where to put it?

Reply:
Why do you need it exactly? I couldn't see it

Reply:
52ee9d9 I moved the bound to serde bounds which was necessary otherwise the derive of serde traits was failing

Reply:
Can you try to implement `Serialize` by hand? This is really weird, there is no reason for `serde` to require `Clone` here :thinking:

Reply:
Implementing `Serialize` manually help, but it didn't for `Deserialize`